ui/pages/confirmations/hooks/useSmartTransactionFeatureFlags.ts

```diff
@@ -0,0 +1,53 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { useEffect } from 'react';
+import { TransactionMeta } from '@metamask/transaction-controller';
+import log from 'loglevel';
+import {
+  getCurrentChainSupportsSmartTransactions,
+  getSmartTransactionsPreferenceEnabled,
+} from '../../../../shared/modules/selectors';
+import { fetchSwapsFeatureFlags } from '../../swaps/swaps.util';
+import {
+  fetchSmartTransactionsLiveness,
+  setSwapsFeatureFlags,
+} from '../../../store/actions';
+import { useConfirmContext } from '../context/confirm';
+
+export function useSmartTransactionFeatureFlags() {
+  const dispatch = useDispatch();
+  const { currentConfirmation } = useConfirmContext<TransactionMeta>();
+  const { id: transactionId, txParams } = currentConfirmation ?? {};
+  const isTransaction = Boolean(txParams);
+
+  const smartTransactionsEnabled = useSelector(
```

Comment:
nit: can we rename it to `const smartTransactionsPreferenceEnabled`? The reason is that we have a function called `getSmartTransactionsEnabled` ([here](https://github.com/MetaMask/metamask-extension/blob/develop/shared/modules/selectors/smart-transactions.ts#L164)) and that one is checking multiple things (feature flags, supported account and chain, etc.), so it would be good to differentiate them.

Reply:
Done.